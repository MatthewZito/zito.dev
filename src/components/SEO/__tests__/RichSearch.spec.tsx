import {
	getAuthorMetadata,
	getOrgMetadata,
	getPostMetadata,
	RichSearchTags
} from '../RichSearch';

import {
	tagListHasEmptyValues,
	sampleSeoData,
	containsEmptyValues
} from './utils';

describe('seo module RichSearchTags', () => {
	it('generates correct tags for pages', () => {
		const generatedTags = RichSearchTags({
			seoData: sampleSeoData.seoSite,
			userData: sampleSeoData.user,
			orgData: sampleSeoData.organization
		});

		expect(generatedTags).toMatchSnapshot();
	});

	it('generates correct tags for posts', () => {
		const generatedTags = RichSearchTags({
			seoData: sampleSeoData.seoPost,
			postData: sampleSeoData.post,
			userData: sampleSeoData.user,
			orgData: sampleSeoData.organization
		});

		expect(generatedTags).toMatchSnapshot();
	});

	it('does not generate tags when missing post `coverImageUrl`', () => {
		const generatedTags = RichSearchTags({
			seoData: sampleSeoData.seoPost,
			postData: { ...sampleSeoData.post, coverImageUrl: undefined },
			userData: sampleSeoData.user,
			orgData: sampleSeoData.organization
		});
		expect(generatedTags).toStrictEqual([]);
	});

	it('does not generate tags when missing post `description`', () => {
		const generatedTags = RichSearchTags({
			seoData: sampleSeoData.seoPost,
			postData: { ...sampleSeoData.post, description: undefined },
			userData: sampleSeoData.user,
			orgData: sampleSeoData.organization
		});

		expect(generatedTags).toStrictEqual([]);
	});

	it('generates correct tags when missing user metadata', () => {
		const generatedTags = RichSearchTags({
			seoData: sampleSeoData.seoPost,
			postData: sampleSeoData.post,
			userData: undefined,
			orgData: sampleSeoData.organization
		});

		expect(generatedTags).toMatchSnapshot();
	});

	it('generates correct tags when missing org data', () => {
		const generatedTags = RichSearchTags({
			seoData: sampleSeoData.seoPost,
			postData: sampleSeoData.post,
			userData: sampleSeoData.user,
			orgData: undefined
		});

		expect(generatedTags).toMatchSnapshot();
	});

	it('does not generate empty tags', () => {
		const generatedSiteTags = RichSearchTags({
			seoData: sampleSeoData.seoSite,
			userData: sampleSeoData.user,
			orgData: sampleSeoData.organization
		});

		expect(tagListHasEmptyValues(generatedSiteTags)).toBe(false);

		const generatedPostTags = RichSearchTags({
			seoData: sampleSeoData.seoPost,
			postData: sampleSeoData.post,
			userData: sampleSeoData.user,
			orgData: sampleSeoData.organization
		});

		expect(tagListHasEmptyValues(generatedPostTags)).toBe(false);

		const generatedPostTagsWithoutUserData = RichSearchTags({
			seoData: sampleSeoData.seoPost,
			postData: sampleSeoData.post,
			userData: undefined,
			orgData: sampleSeoData.organization
		});

		expect(tagListHasEmptyValues(generatedPostTagsWithoutUserData)).toBe(false);

		const generatedTagsWithoutOrgData = RichSearchTags({
			seoData: sampleSeoData.seoPost,
			postData: sampleSeoData.post,
			userData: sampleSeoData.user,
			orgData: undefined
		});

		expect(tagListHasEmptyValues(generatedTagsWithoutOrgData)).toBe(false);
	});
});

describe('`RichSearchTags` utility `getAuthorMetadata`', () => {
	it('generates valid JSONLD', () => {
		const generatedTags = getAuthorMetadata(sampleSeoData.user);

		expect(containsEmptyValues(generatedTags)).toBe(false);

		expect(() => JSON.stringify(generatedTags)).not.toThrow(Error);
	});
});

describe('`RichSearchTags` utility `getOrgMetadata`', () => {
	it('generates valid JSON+LD', () => {
		const generatedTags = getOrgMetadata(sampleSeoData.organization);

		expect(containsEmptyValues(generatedTags)).toBe(false);

		expect(() => JSON.stringify(generatedTags)).not.toThrow(Error);
	});
});

describe('`RichSearchTags` utility `getPostMetadata`', () => {
	it('generates valid JSONLD', () => {
		const generatedTags = getPostMetadata(
			sampleSeoData.post,
			sampleSeoData.organization,
			sampleSeoData.user
		);

		expect(containsEmptyValues(generatedTags)).toBe(false);

		expect(() => JSON.stringify(generatedTags)).not.toThrow(Error);
	});
});
