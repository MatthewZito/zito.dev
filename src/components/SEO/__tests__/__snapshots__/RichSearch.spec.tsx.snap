// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`seo module RichSearchTags generates correct tags for pages 1`] = `
Array [
  <script
    type="application/ld+json"
  >
    {"@context":"https://schema.org","@type":"Organization","url":"https://zito.dev","name":"","description":"","logo":""}
  </script>,
]
`;

exports[`seo module RichSearchTags generates correct tags for posts 1`] = `
Array [
  <script
    type="application/ld+json"
  >
    {"@context":"http://schema.org","@type":"BlogPosting","image":"/logos/logo-1024.png","url":"/posts/url","headline":"Post title","name":"Post title","description":"Post description.","dateCreated":"2021-12-17T00:00:00.000Z","datePublished":"2021-12-17T00:00:00.000Z","dateModified":"2021-12-20T00:00:00.000Z","author":{"@type":"Person","givenName":"Matthew","familyName":"Zito","email":"exbotanical@gmail.com","address":"United States"},"creator":{"@type":"Person","givenName":"Matthew","familyName":"Zito","email":"exbotanical@gmail.com","address":"United States"},"publisher":{"@context":"https://schema.org","@type":"Organization","url":"https://zito.dev","name":"","description":"","logo":""},"mainEntityOfPage":"True","keywords":["Tag1","Tag2"],"articleSection":"Tech","articleBody":"Post body"}
  </script>,
]
`;

exports[`seo module RichSearchTags generates correct tags when missing org data 1`] = `
Array [
  <script
    type="application/ld+json"
  >
    {"@context":"http://schema.org","@type":"BlogPosting","image":"/logos/logo-1024.png","url":"/posts/url","headline":"Post title","name":"Post title","description":"Post description.","dateCreated":"2021-12-17T00:00:00.000Z","datePublished":"2021-12-17T00:00:00.000Z","dateModified":"2021-12-20T00:00:00.000Z","author":{"@type":"Person","givenName":"Matthew","familyName":"Zito","email":"exbotanical@gmail.com","address":"United States"},"creator":{"@type":"Person","givenName":"Matthew","familyName":"Zito","email":"exbotanical@gmail.com","address":"United States"},"mainEntityOfPage":"True","keywords":["Tag1","Tag2"],"articleSection":"Tech","articleBody":"Post body"}
  </script>,
]
`;

exports[`seo module RichSearchTags generates correct tags when missing user metadata 1`] = `
Array [
  <script
    type="application/ld+json"
  >
    {"@context":"http://schema.org","@type":"BlogPosting","image":"/logos/logo-1024.png","url":"/posts/url","headline":"Post title","name":"Post title","description":"Post description.","dateCreated":"2021-12-17T00:00:00.000Z","datePublished":"2021-12-17T00:00:00.000Z","dateModified":"2021-12-20T00:00:00.000Z","publisher":{"@context":"https://schema.org","@type":"Organization","url":"https://zito.dev","name":"","description":"","logo":""},"mainEntityOfPage":"True","keywords":["Tag1","Tag2"],"articleSection":"Tech","articleBody":"Post body"}
  </script>,
]
`;
